### PR TITLE
[DOCS] Copies ESMS monitoring details to Elasticsearch Reference

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/issues/46718

This PR moves the lingering ESMS monitoring page from the Stack Overview (https://www.elastic.co/guide/en/elastic-stack-overview/master/esms.html) to the appropriate section of the Elasticsearch Reference (https://www.elastic.co/guide/en/elasticsearch/reference/master/monitor-elasticsearch-cluster.html).

Preview: http://elasticsearch_48594.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esms.html

### CODE REVIEW

File: docs/reference/monitoring/esms.asciidoc [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+provide better and faster incident resolution.
+
+NOTE: You must obtain your {esms-init} cluster URLs and credentials from the
+Elastic support team.
+
+[discrete]
+[[esms-elasticsearch]]
+=== Collecting monitoring data about {es}
+
+There are two methods for collecting and sending data about the health of your
```
Comment (cachedout) [inline docs/reference/monitoring/esms.asciidoc]: My main questions is whether or not we want to continue to document internal collection as an option at all. We definitely want people using Metricbeat by default at this point and I'm not sure that our customers would be well-served by presenting these two approaches as equivalent options. Thoughts?

Comment (lcawl) [reply docs/reference/monitoring/esms.asciidoc]: I agree that this should be revisited/removed when we clean up the internal collection methods (e.g. in https://github.com/elastic/elasticsearch/pull/47818).  I suggest that we move the content as-is and backport it appropriately, then perform the clean up on the appropriate subset of branches.

Comment (cachedout) [reply docs/reference/monitoring/esms.asciidoc]: Works for me. Thanks, @lcawl

File: docs/reference/monitoring/esms.asciidoc [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+production cluster to {esms-init}: {metricbeat} or collectors and exporters.
```
[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/monitoring/esms.asciidoc]: Good point.  I've converted it to a bulleted list.

File: docs/reference/monitoring/esms.asciidoc [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+--
+
+. {metricbeat-ref}/metricbeat-starting.html[Start {metricbeat}]. 
+
+. Verify that your monitoring data exists in {esms-init}.
++
+--
+Open {kib} in your web browser. Use the {kib} URL and the administrator user ID
+that was provided to you by the Elastic support team.
+{kibana-ref}/kibana-page.html[View the {kib} metrics] on the *Stack Monitoring* page.
```
[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/monitoring/esms.asciidoc]: Good catch, thanks!

File: docs/reference/monitoring/esms.asciidoc [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+(`metricbeat.yml`):
+
+[source,yaml]
+----------------------------------
+output.elasticsearch:
+  hosts: ["MONITORING_ELASTICSEARCH_URL"] <1>
+  username: cloud_monitoring_agent <2>
+  password: MONITORING_AGENT_PASSWORD <3>
+----------------------------------
+<1> Replace `MONITORING_ELASTICSEARCH_URL` with the appropriate URL for {esms-init}.
```
[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/monitoring/esms.asciidoc]: Done, thanks!

**Pull request merged**